pyproject.toml

```diff
@@ -46,24 +46,25 @@ classifiers = [
     "Topic :: Scientific/Engineering :: Visualization",
 ]
 dependencies = [
-    "anndata>=0.7.4",
+    "anndata>=0.8",
     # numpy needs a version due to #1320
-    "numpy>=1.17.0",
+    "numpy>=1.23",
     "matplotlib>=3.6",
-    "pandas >=2.1.3",
-    "scipy>=1.4",
-    "seaborn>=0.13.0",
-    "h5py>=3",
+    "pandas >=1.5",
+    "scipy>=1.8",
+    "seaborn>=0.13",
+    "h5py>=3.1",
     "tqdm",
     "scikit-learn>=0.24",
-    "statsmodels>=0.10.0rc2",
+    "statsmodels>=0.13",
     "patsy",
-    "networkx>=2.3",
+    "networkx>=2.7",
     "natsort",
     "joblib",
-    "numba>=0.41.0",
+    "numba>=0.56",
     "umap-learn>=0.3.10",
```

Comment:
Should this be bumped to 0.5, as requested in https://github.com/scverse/scanpy/issues/2012?

Reply:
I would like that, but would postpone to a follow up PR. IIRC there's significant backwards compatibility code that should be removed at the same time, which would take some investigating. I think there's a lot of value getting this PR in ASAP due to the amount of bugs it's caught so far.

Reply:
Would you be up for looking into that? It seems quite related to some recent work you've done around the neighbors code

Reply:
OK!